readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/offset/HdfsOffsetComputer.java

```diff
@@ -45,51 +50,107 @@ public HdfsOffsetComputer(FileSystem fs, Path basePath, @Nullable String kafkaCl
+            result.put(partitionIdDateFile.getKey(), getMaxOffset(partitionIdDateFile.getValue()));
+        }
+
         return result;
     }
 
-    private String computeDirNamesPattern(LocalDateTime today, int backlogDays) {
-        List<String> dirnames = new ArrayList<>(backlogDays);
+    private boolean isPartitionComputedByThisReader(int partitionId, Set<Integer> partitionIdsSet) {
+        return partitionIdsSet.contains(partitionId);
+    }
+
+    private boolean chekCurrentFileIndexBiggerTheOneInHashPartitionDay(int partitionId, String listedDay, long index, Map<Integer,
+        Map<String, FinalEventPartitionFile>> resultFile) {
+        return isPartitionDayNotAlreadyInHashPartitionDay(partitionId, listedDay, resultFile) || index > resultFile.get(partitionId).get(listedDay).getIndex();
+    }
 
-        for (int i = 0; i < backlogDays; ++i) {
-            LocalDateTime day = today.minusDays(i);
+    private boolean isPartitionDayNotAlreadyInHashPartitionDay(int partitionId, String listedDay, Map<Integer,
+        Map<String, FinalEventPartitionFile>> resultFile) {
+        return !(resultFile.containsKey(partitionId) && resultFile.get(partitionId).containsKey(listedDay));
+    }
 
-            dirnames.add(computeDirName(day));
-        }
+    protected Long getMaxOffset(Map<String, FinalEventPartitionFile> dateFinalEventPartitionFile) {
+        List<Long> maxOffsets = new ArrayList<>();
+        dateFinalEventPartitionFile.values().stream()
+            .forEach(finalEventPartitionFile -> {
+                // Read parquet file
+                ParquetFileReader pFR;
+                try {
+                    pFR = ParquetFileReader.open(fs.getConf(), finalEventPartitionFile.getFilePath());
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+
+                // Get max value from all blocks
+                long maxValue = pFR.getFooter().getBlocks().stream()
+                    .map(b -> {
+                        Optional<ColumnChunkMetaData> columnChunkMetaData1 = b.getColumns().stream()
+                            .filter(column -> Arrays.stream(column.getPath().toArray()).allMatch(path -> path.equals("kafka_offset")))
+                            .findFirst();
+
+                        if (columnChunkMetaData1.isPresent()) {
+                            return ((LongStatistics) columnChunkMetaData1.get().getStatistics()).genericGetMax();
+                        } else {
+                            return NO_OFFSET;
+                        }
+
+                    })
+                    .reduce(NO_OFFSET, (max1, max2) -> max1 > max2 ? max1 : max2);
+
+                maxOffsets.add(maxValue);
+            });
+
+        // Get max offset from all files for a partition
+        return maxOffsets.stream().mapToLong(Long::longValue).max().orElse(NO_OFFSET);
```

Comment:
Instead of populating a list and then stream on it to find max, you can use directly the stream dateFinalEventPartitionFile.values().stream() at line 122